update "First Steps" tutorial (#553)
SwissDataScienceCenter/renku: docs/user/firststeps.rst

or LinkedIn account by clicking on the corresponding button.

Once logged in, create a new project by going to the **Projects** (1) page
and clicking on the **New Project** (2) button.

.. image:: ../_static/images/ui_create_project.png
    :width: 100%
    :align: center
    :alt: Head to new project page

Set **Zurich bikes tutorial** as your project title (1), fill-in a short
description (2) and set the project visibility to **Public** (3). Click on the
**Create** button (4).

.. image:: ../_static/images/ui_new_project.png
    :width: 100%
    :align: center
    :alt: Create a new project

Now that we have a project, we can start working on it by starting a

> new JupyterLab. Click on **Notebook Servers** (1), then on

Reviewer: ```suggestion
new JupyterLab notebook server. Click on **Notebook Servers** (1), then on
```

**Start new server** (2).

.. image:: ../_static/images/ui_notebook_servers.png
    :width: 100%
    :align: center
    :alt: Head to Notebook ui_notebook_servers

> The default settings are fine, click on **Launch Server**. You will see

Reviewer: ```suggestion
The default settings are fine for this tutorial, just click on **Launch Server**. You will see
```

> a new row with a yellow button on the right. Wait until its color turns

Reviewer: ```suggestion
a new row with a yellow button on the right. Wait until its color has turned
```

> to green and the text from **Starting** to **Running**

Reviewer: ```suggestion
to green and the text from **Starting** to **Running**.
```

.. note::

    Please be patient, the first time you start a server it may require

>     up to one minute

Reviewer: ```suggestion
    up to one minute.
```

You can now connect to the server by clicking on the green button (1),
then on **Connect** (2)

.. image:: ../_static/images/ui_connect_to_server.png
    :width: 100%
    :align: center
    :alt: Start a new server

.. _add_data:

> Adding data to your project

Reviewer: ```suggestion
Add data to your project
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^

In the JupyterLab interface, we can see that a few files already exist.
Let's start by adding data using the `Renku CLI <https://renku-python.readthedocs.io/en/latest/cli.html>`__.

> From JupyterLab, start a terminal by clicking in the **Launcher** page on

Reviewer: I suggest:
```
From JupyterLab, start a terminal by clicking the **Terminal** icon (1) on the bottom right of the **Launcher** page.
```

    # Author: John Doe <john.doe@example.com>
    # Date:   Mon Apr 29 11:56:30 2019 +0000
    #
    #     renku dataset create zhbikes
    #
    # commit 3f74a2dfdf5e27c1dc124f6455931089023253b8
    # Author: John Doe <john.doe@example.com>
    # Date:   Mon Apr 29 11:53:41 2019 +0000
    #
    #     init renku repository

.. code-block:: console

    git status
    # Output similar to:
    # On branch master
    # Your branch is ahead of 'origin/master' by 2 commits.
    #   (use "git push" to publish your local commits)
    #
    # nothing to commit, working directory clean

Let's push the two fresh commits by running:

.. code-block:: console

    git push
    # Output similar to:
    # Locking support detected on remote "origin". Consider enabling it with: [...]
    # Counting objects: 19, done. (1/1), 66 MB | 0 B/s
    # Delta compression using up to 8 threads.
    # Compressing objects: 100% (15/15), done.
    # Writing objects: 100% (19/19), 2.26 KiB | 463.00 KiB/s, done.
    # Total 19 (delta 3), reused 0 (delta 0)
    # To https://renkulab.io/gitlab/john.doe/zurich-bikes-tutorial.git
    #     b55aea9..91b226b  master --> master

> The data file can be opened from JupyterLab by going to the **File** tab

Reviewer: ```suggestion
The data file can be opened from JupyterLab by browsing to the **File** tab
```

> on the top left (1) and traversing the ``data`` folder (2).

Reviewer: ```suggestion
on the top left (1) and clicking the ``data`` folder (2).
```

Author: changed slightly since the file is in a sub folder

Reviewer: ok

.. _python_environment:

> Installing and managing Python packages

Reviewer: I'd stick to the original formulation here.

.. _jupyterlab:

> Working with Renku within JupyterLab

Reviewer: ```suggestion
Use Renku from within JupyterLab
```

.. _interactive_exploration:

> Interactively exploring the bicycle counting data

Reviewer: ```suggestion
Interactively explore the bicycle count data
```

    git add src
    git commit -m "added refactored scripts"
    git push
    # [...]
    # To https://dev.renku.ch/gitlab/john.doe/zurich-bikes-tutorial.git
    #     a40f192..7922ee1  master -> master

.. _create_workflow:

> Producing a repeatable analysis

Reviewer: I'd stick to the original here

python.readthedocs.io/en/latest/cli.html>`_ for more information on this and
other commands.

.. note::

    Did you get an error like this?

    .. code-block:: console

        # Traceback (most recent call last):
        #   File "src/clean_data.py", line 72, in <module>
        #     clean_data(args.input, args.output)
        # [...]
        # KeyError: "None of [Index(['fk_zaehler', 'datum', 'velo_in', 'velo_out', 'fuss_in', 'fuss_out',\n'objectid'],\n      dtype='object')] are in the [columns]"
        # Error: Command returned non-zero exit status 1.

    If you played around with the tutorial and you started a new
    Jupyter server, this may happen. Why?
    `Under the hood <https://renku-python.readthedocs.io/en/latest/api.html>`_,
    we use
    `git-lfs <https://git-lfs.github.com/>`_
    to save big files and we automatically fetch them only
    when it is required by a `renku` command. If you check the
    ``2019_verkehrszaehlungen_werte_fussgaenger_velo.csv`` file you
    will see only a few lines of metadata starting with

>     ``version https:://fit-lfs.github.com/spec/v1``. You can easily

Reviewer: ```suggestion
    ``version https://git-lfs.github.com/spec/v1``. You can easily
```

>     fetch the data manually by pulling from the console

Reviewer: ```suggestion
    fetch the data manually from the console by running 
```

.. code-block:: console

    git push
    # [...]
    # Uploading LFS objects: 100% (7/7), 69 MB | 25 MB/s, done
    # Counting objects: 39, done.
    # Delta compression using up to 8 threads.
    # Compressing objects: 100% (36/36), done.
    # Writing objects: 100% (39/39), 4.59 KiB | 1.15 MiB/s, done.
    # Total 39 (delta 14), reused 0 (delta 0)
    # To https://dev.renku.ch/gitlab/lorenzo.cavazzi.tech/deleteme.git
    #    8892173..8d00b71  master -> master

.. _sharing_is_caring:

> Sharing your results and collaborating with your peers

Reviewer: Again, I'd revert to the original formulation.
